### PR TITLE
feat(DDI-1001): Adding form item examples

### PR DESCRIPTION
**Pull request by Radha-Alberta**
Removed Form Row example.
Added examples of the form item wrapping: Radio, Checkbox, Dropdown, Button

### CODE REVIEW

File: libs/docs/src/components/common/FormItem.stories.mdx [hunk 1]
```diff
@@ -51,11 +56,36 @@ import {
...
   <GoAFormItem label="First name">
-    <GoAInput name="firstName" onChange={() => {}} />
+    <GoAInput name="firstName"/>
+  </GoAFormItem>
+  <div style={{ padding: '2px'}}></div>
+  <GoAFormItem label="Checkbox Colors">
+    <GoACheckbox name="colors" text="Red" value="red"></GoACheckbox>
+    <GoACheckbox name="colors" text="Blue" value="blue"></GoACheckbox>
+  </GoAFormItem>
+  <div style={{ padding: '2px'}}></div>
```
Comment (chrisolsen) [inline libs/docs/src/components/common/FormItem.stories.mdx:70]: Not sure we should be adding additional padding for the reason that people will notice that the same padding isn't attained when they use this component in their examples.

File: libs/docs/src/components/common/FormItem.stories.mdx [hunk 2]
```diff
@@ -84,19 +152,65 @@ import {
...
+    </GoARadioGroup>
+  </GoAFormItem>
+  <div style={{ padding: '2px'}}></div>
+  <GoAFormItem label="Dropdown Colors" optional>
+    <GoADropdown name="colors">
+      <GoADropdownOption name="colors" value="red" label="Red" />
+      <GoADropdownOption name="colors" value="green" label="Green" />
+    </GoADropdown>
+  </GoAFormItem>
+  <div style={{ padding: '2px'}}></div>
```
Comment (chrisolsen) [inline libs/docs/src/components/common/FormItem.stories.mdx:209]: same thing here with the padding

File: libs/docs/src/components/common/FormItem.stories.mdx [hunk 3]
```diff
@@ -166,22 +402,65 @@ import {
...
+## Error text
 
-<Story name="Required">
+<Story name="Error text">
   <GoAFormItem label="First name" error="First name is required">
-    <GoAInput name="firstName" onChange={() => {}} error />
+    <GoAInput name="firstName" error />
+  </GoAFormItem>
+  <div style={{ padding: '2px'}}></div>
+  <GoAFormItem label="Checkbox Colors" error="Atleast one option is required">
```
Comment (chrisolsen) [inline libs/docs/src/components/common/FormItem.stories.mdx:441]: Missing spacing between `At` and `least`

File: libs/docs/src/components/common/FormItem.stories.mdx [hunk 4]
```diff
@@ -191,105 +470,89 @@ import {
...
         </GoAFormItem>
-      </GoAFlexRow>
-    `}
-    />
+          <GoAFormItem label="Button label" error="Click here for more">
+          <GoAButtonGroup alignment="start">
+            <GoAButton type="primary">Next</GoAButton>
+          </GoAButtonGroup>
+        </GoAFormItem>
+      `}/>
```
Comment (chrisolsen) [inline libs/docs/src/components/common/FormItem.stories.mdx:555]: I don't think this component should ever be used with a button.

Comment (chrisolsen) [reply libs/docs/src/components/common/FormItem.stories.mdx:555]: After talking to Tom I guess there is a scenario when it maybe used.

File: libs/docs/src/components/common/FormItem.stories.mdx (hunk 1 above)
Comment (ArakTaiRoth) [reply libs/docs/src/components/common/FormItem.stories.mdx:70]: Change to 8px everywhere

Comment (Radha-Alberta) [reply libs/docs/src/components/common/FormItem.stories.mdx]: Done

File: libs/docs/src/components/common/FormItem.stories.mdx (hunk 2 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/FormItem.stories.mdx]: Changed to 8px

File: libs/docs/src/components/common/FormItem.stories.mdx (hunk 3 above)
Comment (Radha-Alberta) [reply libs/docs/src/components/common/FormItem.stories.mdx]: Added spacing

File: libs/docs/src/components/common/FormItem.stories.mdx [hunk 5]
```diff
@@ -191,105 +470,89 @@ import {
...
-        <!-- Error message should be a dynamic value -->
-        <GoAFormItem label="First name" error="First name is required" helpText="Help text">
-          <GoAInput name="firstName" onChange={() => {}} error />
+        <GoAFormItem label="First name" error="First name is required">
+          <GoAInput name="firstName" value="" onChange={() => void 0} error />
         </GoAFormItem>
-        <!-- Blank error indicates a valid field -->
-        <GoAFormItem label="Favourite Colour" error="">
-          <GoAInput name="favcolor" onChange={() => {}} />
+        <GoAFormItem label="Checkbox Colors" error="Atleast one option is required">
```
[the remark this answers is not in the document]
Comment (Radha-Alberta) [reply libs/docs/src/components/common/FormItem.stories.mdx]: Added spacing

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**